packages/inventory/src/containers/InventoryAddEdit/index.tsx

```diff
@@ -0,0 +1,267 @@
+  locationUnitsReducer,
+} from '@opensrp/location-management';
+import {
+  LOCATIONS_GET_ALL_SYNC_ENDPOINT,
+  OPENSRP_ENDPOINT_GET_INVENTORIES,
+  ROUTE_PARAM_INVENTORY_ITEM_ID,
+  ROUTE_PARAM_SERVICE_POINT_ID,
+} from '../../constants';
+import {
+  fetchInventories,
+  getInventoriesByIdsFactory,
+  Inventory,
+  inventoryReducer,
+  inventoryReducerName,
+} from '../../ducks/inventory';
+import { InventoryItemForm, defaultInitialValues } from '../../components/InventoryItemForm';
+
+/** register reducers */
+reducerRegistry.register(locationUnitsReducerName, locationUnitsReducer);
+reducerRegistry.register(productCatalogueReducerName, ProductCatalogueReducer);
+reducerRegistry.register(inventoryReducerName, inventoryReducer);
+
+/** Route params */
+interface RouteParams {
+  [ROUTE_PARAM_SERVICE_POINT_ID]: string;
+  [ROUTE_PARAM_INVENTORY_ITEM_ID]?: string;
+}
+
+/** interface component props */
+export interface InventoryAddEditProps extends RouteComponentProps<RouteParams> {
+  products: ProductCatalogue[];
+  openSRPBaseURL: string;
+  cancelURL: string;
+  redirectURL: string;
+  fetchProductsCreator: typeof fetchProducts;
+  fetchLocationUnitsCreator: typeof fetchLocationUnits;
+  fetchInventoriesCreator: typeof fetchInventories;
+  servicePoint: LocationUnit | null;
+  inventory?: Inventory | null;
+  customFetchOptions?: typeof getFetchOptions;
+}
+
+/** default component props */
+export const defaultInventoryAddEditProps = {
+  openSRPBaseURL: '',
+  cancelURL: '',
+  redirectURL: '',
+  products: [],
+  fetchProductsCreator: fetchProducts,
+  fetchLocationUnitsCreator: fetchLocationUnits,
+  fetchInventoriesCreator: fetchInventories,
+  servicePoint: null,
+};
+
+const InventoryAddEdit: React.FC<InventoryAddEditProps> = (props: InventoryAddEditProps) => {
+  const {
+    openSRPBaseURL,
+    cancelURL,
+    products,
+    fetchProductsCreator,
+    fetchLocationUnitsCreator,
+    fetchInventoriesCreator,
+    redirectURL,
+    servicePoint,
+    inventory,
+    match,
+    customFetchOptions,
+  } = props;
+  const [UNICEFSections, setUNICEFSections] = React.useState<Setting[]>([]);
+  const [donors, setDonors] = React.useState<Setting[]>([]);
+  const { Title } = Typography;
+  const isEdit = !!match.params[ROUTE_PARAM_INVENTORY_ITEM_ID];
+
+  useEffect(() => {
+    // Handle when servicePoint is null e.g when a user refreshes page
+    if (!servicePoint) {
+      // fetch location units
+      const service = new OpenSRPService(
+        LOCATIONS_GET_ALL_SYNC_ENDPOINT,
+        openSRPBaseURL,
+        customFetchOptions
+      );
+      service
+        .list({
+          serverVersion: 0,
+          // eslint-disable-next-line @typescript-eslint/camelcase
+          is_jurisdiction: false,
+        })
+        .then((response: LocationUnit[]) => {
+          fetchLocationUnitsCreator(response);
+        })
+        .catch((_: HTTPError) => {
+          sendErrorNotification(ERROR_GENERIC);
+        });
+    }
```

Comment:
this seems to be missing a limit, 
also do we need the whole service points lot, wouldn't a request to jest get the one specific location work here.